### PR TITLE
Add experimental Multi DC capabillity of Scylla cluster deployment

### PR DESCRIPTION
**Pull request by zimnx**
Adds new CRDs and extends Operator to be able to deploy multi DC ScyllaDB clusters within external Kubernetes clusters.

The **v1alpha1.RemoteKubernetesCluster** API is a new addition to the Scylla Operator, designed to enhance multi-datacenter capabilities for Scylla clusters. 
This API serves as an abstraction for remote Kubernetes clusters, allowing the Scylla Operator to manage resources across multiple Kubernetes clusters. 
By leveraging this API, the Operator can reconcile resources not only in the local Kubernetes cluster but also in remote ones, facilitating the deployment and management of Scylla clusters across diverse environments.

The **v1alpha1.ScyllaDBCluster** API is an extension of the existing v1.ScyllaCluster API in the Scylla Operator, introducing multi-datacenter capabilities to Scylla deployments. 
This new API allows users to define and manage a Scylla cluster spanning multiple datacenters, including those in remote Kubernetes clusters. 
When used alongside the v1alpha1.RemoteKubernetesCluster API, users can specify remote Kubernetes clusters for each datacenter, enabling the Operator to reconcile and manage Scylla datacenters in geographically dispersed locations.
This advancement simplifies the creation and management of globally distributed Scylla deployments, ensuring consistent operations and high availability across multiple datacenters.

These APIs are experimental, some features compared to existing ScyllaCluster may be missing or be broken, so take great care when using it.

### CODE REVIEW

**Comment by rzetelskik** [comment]
@zimnx would you be able to say what changed compared to enterprise repo and so requires extra attention?

**Comment by zimnx** [comment]
> @zimnx would you be able to say what changed compared to enterprise repo and so requires extra attention?

not much, mainly names to get rid of `enterprise` suffix and couple of type structures were renamed to avoid collisions in v1alpha1 group.

**Comment by tnozicka** [comment]
Shouldn't we ship this with the controllers at once? If we end up in the middle this effectively blocks making any release.

**Comment by zimnx** [comment]
> Shouldn't you ship this with the controllers at once? If we end up in the middle this effectively blocks making any release.

I'm not sure what you mean. This essentially merges everything we have in other repo into this one in one PR.

File: deploy/operator/00_operator_remote.clusterrole_def.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
[the remark this answers is not in the document]
Comment (rzetelskik) [reply deploy/operator/00_operator_remote.clusterrole_def.yaml:1]: (applies to all in deploy/operator)

**Review by rzetelskik — COMMENTED** [review]
a couple of comments, rest lgtm

File: test/e2e/fixture/scylla/scylladbcluster.yaml.tmpl [hunk 2]
```diff
@@ -0,0 +1,41 @@
+apiVersion: scylla.scylladb.com/v1alpha1
```
[the remark this answers is not in the document]
Comment (zimnx) [reply test/e2e/fixture/scylla/scylladbcluster.yaml.tmpl:1]: Currently we don't have separate directories for different versions in test fixture

File: deploy/operator/00_operator_remote.clusterrole_def.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (zimnx) [reply deploy/operator/00_operator_remote.clusterrole_def.yaml:1]: it is, moved to autogenerated commit

File: deploy/operator/00_operator_remote.clusterrole.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
[the remark this answers is not in the document]
Comment (zimnx) [reply deploy/operator/00_operator_remote.clusterrole.yaml:1]: it is, moved to autogenerated commit

**Comment by tnozicka** [comment]
> I'm not sure what you mean. This essentially merges everything we have in other repo into this one in one PR.

ok, then the PR title and description could use some tuning.

The PR description should be more descriptive in general. Given how large the change is, it fails to mention e.g. any of the new APIs being introduced.

> This feature is experimental, we don't provide any gurantees.

I suppose this could say something like: `ScyllaDBCluster` in group `scylla.scylladb.com/v1alpha1` is experimental and should be treated with care.

> New APIs may be changed **without keeping backward compatibility.**

I don't get this. The API rules are clear and applicable to all resources, always. Backwards compatibility for v1alpha1 resources is guaranteed, but we can replace it with v1alpha2 without a transition period.
(Note that the compatibility rules, by definition apply to released versions, so a lot can happen until v1.16.0.)

**Review by tnozicka — APPROVED** [review]
/approve

**Review by rzetelskik — APPROVED** [review]
/lgtm
thanks @zimnx

**Pull request merged**